runtime/metricsview/executor_validate.go

```diff
@@ -302,10 +309,41 @@ func validateDimension(ctx context.Context, olap drivers.OLAPStore, t *drivers.T
 }
 
 // validateMeasure validates a metrics view measure.
-func validateMeasure(ctx context.Context, olap drivers.OLAPStore, t *drivers.Table, m *runtimev1.MetricsViewSpec_MeasureV2) error {
-	err := olap.Exec(ctx, &drivers.Statement{
-		Query:  fmt.Sprintf("SELECT 1, (%s) FROM %s GROUP BY 1", m.Expression, olap.Dialect().EscapeTable(t.Database, t.DatabaseSchema, t.Name)),
+func (e *Executor) validateMeasure(ctx context.Context, t *drivers.Table, m *runtimev1.MetricsViewSpec_MeasureV2) error {
+	err := e.olap.Exec(ctx, &drivers.Statement{
+		Query:  fmt.Sprintf("SELECT 1, (%s) FROM %s GROUP BY 1", m.Expression, e.olap.Dialect().EscapeTable(t.Database, t.DatabaseSchema, t.Name)),
 		DryRun: true,
 	})
 	return err
 }
+
+// validateSchema validates that the metrics view's measures are numeric.
+func (e *Executor) validateSchema(ctx context.Context, res *ValidateMetricsViewResult) error {
+	// Resolve the schema of the metrics view's dimensions and measures
+	schema, err := e.Schema(ctx)
+	if err != nil {
+		return err
+	}
+	types := make(map[string]*runtimev1.Type, len(schema.Fields))
+	for _, f := range schema.Fields {
+		types[f.Name] = f.Type
+	}
+
+	// Check that the measures are not strings
+	for i, m := range e.metricsView.Measures {
+		typ, ok := types[m.Name]
+		if !ok {
+			return fmt.Errorf("internal: measure %q not found in schema", m.Name)
+		}
+
+		switch typ.Code {
+		case runtimev1.Type_CODE_TIMESTAMP, runtimev1.Type_CODE_DATE, runtimev1.Type_CODE_TIME, runtimev1.Type_CODE_STRING, runtimev1.Type_CODE_BYTES, runtimev1.Type_CODE_ARRAY, runtimev1.Type_CODE_STRUCT, runtimev1.Type_CODE_MAP, runtimev1.Type_CODE_JSON, runtimev1.Type_CODE_UUID:
```

Comment:
Not sure if could happen but should we also include `CODE_UNSPECIFIED` as well ? Also I think min and max like functions can be used on string columns on duckdb and return string types but I don't see obvious usage for them.